### PR TITLE
Add auto materialize policy to gql

### PR DESCRIPTION
**Pull request by johannkm**
Initial backend for https://github.com/dagster-io/dagster/issues/13446

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13774** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13774" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13774?utm_source=stack-comment).

File: python_modules/dagster-graphql/dagster_graphql/schema/auto_materialize_policy.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+class GrapheneAssetFreshnessInfo(graphene.ObjectType):
+    currentMinutesLate = graphene.Field(graphene.Float)
+    latestMaterializationMinutesLate = graphene.Field(graphene.Float)
+
+    class Meta:
+        name = "AssetFreshnessInfo"
+
+
+class GrapheneFreshnessPolicy(graphene.ObjectType):
```
Comment (OwenKephart) [inline python_modules/dagster-graphql/dagster_graphql/schema/auto_materialize_policy.py:18]: forgot to remove? :P

File: python_modules/dagster/dagster/_core/definitions/auto_materialize_policy.py [hunk 2]
```diff
@@ -52,19 +59,32 @@ def time_window_partition_scope(self) -> Optional[datetime.timedelta]:
...
-        return AutoMaterializePolicy(
-            on_missing=True,
-            on_new_parent_data=False,
-            for_freshness=True,
-            time_window_partition_scope_minutes=datetime.timedelta.resolution.total_seconds() / 60,
-        )
+        return _LAZY_POLICY
+
+    @property
+    def policy_type(self) -> AutoMaterializePolicyType:
```
Comment (OwenKephart) [inline python_modules/dagster/dagster/_core/definitions/auto_materialize_policy.py:70]: I think I'd prefer if, for now, we just made this a function of `self.on_new_parent_data`. I.e. if that's true, you're eager, if it's false, you're lazy. We can add some code in the __new__ method on this class to enforce that you never have `on_new_parent_data` and `for_freshness` as both false (meaning `not on_new_parent_data` would imply `for_freshness`, which would mean that you're lazy)

File: python_modules/dagster/dagster/_core/definitions/auto_materialize_policy.py [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
 from typing import NamedTuple, Optional
 
 from dagster._annotations import experimental, public
 from dagster._serdes.serdes import whitelist_for_serdes
 
 
+class AutoMaterializePolicyType(Enum):
+    EAGER = "EAGER"
+    LAZY = "LAZY"
+    CUSTOM = "CUSTOM"
```
Comment (OwenKephart) [inline python_modules/dagster/dagster/_core/definitions/auto_materialize_policy.py:12]: see below, but I don't think we should inclue `CUSTOM` until we are sure we need it

**Review by OwenKephart — CHANGES_REQUESTED** [review]

File: python_modules/dagster/dagster/_core/definitions/auto_materialize_policy.py (hunk 2 above)
Comment (johannkm) [reply python_modules/dagster/dagster/_core/definitions/auto_materialize_policy.py:70]: that seems reasonable... how would you make the distinction once we have on_new_code_version?

Comment (johannkm) [reply python_modules/dagster/dagster/_core/definitions/auto_materialize_policy.py:106]: Gotcha, so lazy = `for_freshness` and not (`on_new_parent_data` or `on_new_code_version`). eager = `on_new_parent_data` or `on_new_code_version`, `for_freshness` can be on or off

**Review by OwenKephart — APPROVED** [review]
nice

**Pull request merged**